src/core/kernel/cpulevel.cpp

```diff
@@ -0,0 +1,51 @@
+/*
+* Copyright (c) 2012-2019 Fredrik Mellbin
+*
+* This file is part of VapourSynth.
+*
+* VapourSynth is free software; you can redistribute it and/or
+* modify it under the terms of the GNU Lesser General Public
+* License as published by the Free Software Foundation; either
+* version 2.1 of the License, or (at your option) any later version.
+*
+* VapourSynth is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+* Lesser General Public License for more details.
+*
+* You should have received a copy of the GNU Lesser General Public
+* License along with VapourSynth; if not, write to the Free Software
+* Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+*/
+
+#include <cstring>
```

Comment:
Shouldn't this be a **C** file?  
(see associated header)

Reply:
What for?

Reply:
* Header defines functions in C context
* File contains solely C code (f.e. `strcmp`)

Reply:
`core->getCpuLevel()` is a C++ statement...

Reply:
Oh, i didn't see **VSCore** isn't a struct holding function pointers but a public C++ class, sorry. Was already wondering why these thin wrapper functions exist anyway.

Found this C & C++ mix strange. `!strcmp(...)` always looks like the opposite expectation, using [std::string == operator](https://en.cppreference.com/w/cpp/string/basic_string/operator_cmp) looks more logical and already stores string length for faster compare.